### PR TITLE
feat(promisify): use native promisify and remove es6-promisify

### PR DESCRIPTION
**Pull request by rxmarbles**
Remove unnecessary package as we can use native promisify. Also switched ot using `Promis.allSettled` for each of the processes in `cleanup`

### CODE REVIEW

File: lib/terminus.js [hunk 1]
```diff
@@ -108,10 +106,12 @@ function decorateWithSignalHandler (server, state, options) {
     if (!state.isShuttingDown) {
       state.isShuttingDown = true
       try {
-        await beforeShutdown()
```
Comment (gergelyke) [inline lib/terminus.js:111]: it's imperative that these events are called in sequential order

**Review by gergelyke — CHANGES_REQUESTED** [review]

**Review by gergelyke — APPROVED** [review]
thank you

**Comment by gergelyke** [comment]
it seems like that your code style changes broke the TS type checks - can you look into them please?

**Comment by rxmarbles** [comment]
@gergelyke Not sure what the culprit was. So I reverted pacakge.json and lock file to what is in main and then removed the `es6-promisify` via `npm unistalled` ran it locally and seems to work now.

**Pull request merged**